src/MySqlConnector/Core/ServerSession.cs

```diff
@@ -454,30 +452,26 @@ public async Task DisposeAsync(IOBehavior ioBehavior, CancellationToken cancella
 			}
 
 			ServerVersion = new(initialHandshake.ServerVersion);
-			ConnectionId = initialHandshake.ConnectionId;
+			Context = new Context(initialHandshake.ProtocolCapabilities, cs.Database, initialHandshake.ConnectionId);
```

Comment:
By implementing an interface on `ServerSession` we also avoid this second `Context` allocation every time a session is connected (so reduced from three back down to one object).